server/src/test/java/org/elasticsearch/common/xcontent/support/filtering/FilterPathTests.java

```diff
@@ -295,7 +296,7 @@ public void testContainsDoubleWildcardFilterPath() {
     }
 
     public void testMultipleFilterPaths() {
-        Set<String> inputs = Sets.newHashSet("foo.**.bar.*", "test.dot\\.ted");
+        Set<String> inputs = new LinkedHashSet<>(Arrays.asList("foo.**.bar.*", "test.dot\\.ted"));
 
         FilterPath[] filterPaths = FilterPath.compile(inputs);
```

Comment:
Given that the `inputs` (to FilterPath::compile) is a Set, then the order of the returned filterPaths array should not be assumed. Changing to LinkedHashSet here assumes that compile will follow iteration order. I'm not sure that we want this?

Reply:
I don't think we do. Right now we end up going `String -> List<String> -> Set<String> -> FilterPath[]` and have a fairly randomly sorted array. This is more out of convenience and wanting to remove duplicates than anything. I don't feel like we thought about order. And we probably should continue not thinking about order.

Reply:
It's needed pretty much just for this test, because we check the filter paths by the index, so `foo.bar` has to be at the first position.

Reply:
Its kind of a weird test I guess - everywhere else we use these filter paths we iterate over the array. Maybe this test should do the same. List `assertThat(Arrays.stream(filterPaths).map(m -> m == null).count(), equalTo(0L));`. It's longer but its more true to how we run use the two filter paths in reality.